titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/workflow/DefaultNodeConditionController.java

```diff
@@ -0,0 +1,161 @@
+package com.netflix.titus.supplementary.relocation.workflow;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.titus.api.jobmanager.JobAttributes;
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.JobDescriptor;
+import com.netflix.titus.api.jobmanager.service.ReadOnlyJobOperations;
+import com.netflix.titus.api.model.callmetadata.CallMetadata;
+import com.netflix.titus.api.model.callmetadata.Caller;
+import com.netflix.titus.api.model.callmetadata.CallerType;
+import com.netflix.titus.common.framework.scheduler.ExecutionContext;
+import com.netflix.titus.common.framework.scheduler.ScheduleReference;
+import com.netflix.titus.common.framework.scheduler.model.ScheduleDescriptor;
+import com.netflix.titus.common.runtime.TitusRuntime;
+import com.netflix.titus.common.util.retry.Retryers;
+import com.netflix.titus.runtime.connector.jobmanager.JobDataReplicator;
+import com.netflix.titus.runtime.connector.jobmanager.JobManagementClient;
+import com.netflix.titus.supplementary.relocation.RelocationConfiguration;
+import com.netflix.titus.supplementary.relocation.connector.Node;
+import com.netflix.titus.supplementary.relocation.connector.NodeDataResolver;
+import com.netflix.titus.supplementary.relocation.util.RelocationUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class DefaultNodeConditionController implements NodeConditionController {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultNodeConditionController.class);
+    private final RelocationConfiguration configuration;
+    private final NodeDataResolver nodeDataResolver;
+    private final JobDataReplicator jobDataReplicator;
+    private final ReadOnlyJobOperations jobOperations;
+    private final JobManagementClient jobManagementClient;
+    private final NodeConditionCtrlMetrics metrics;
+    private final TitusRuntime titusRuntime;
+
+    private static final CallMetadata CALL_METADATA = buildCallMetadata();
+    private static final String CALLER_APP_ID = "titusrelocation";
+    private static final String CALL_REASON = "Bad Node Condition";
```

Comment:
I think using a more descriptive reason here would be better like: `This task was automatically terminated because the underlying host had issues.`